source/language/gates.rst

```diff
@@ -240,58 +175,32 @@ and these symbols are scoped only to the subroutine body.
 
 An empty body corresponds to the identity gate.
 
-Gates must be declared before use and
+To avoid infinite recursion, gates must be declared before use and
 cannot call themselves. The statement ``name(params) qargs;`` applies the gate,
-and the variable parameters ``params`` can have any numeric type.
-
-The gate can be applied to any combination of qubit registers *of the same size*, as shown in the following example.
-
-The quantum circuit given by
-
-.. code-block::
-
-   gate g qb0, qb1, qb2, qb3
-   {
-     // body
-   }
-   qubit qr0[1];
-   qubit qr1[2];
-   qubit qr2[3];
-   qubit qr3[2];
-   g qr0[0], qr1, qr2[0], qr3; // ok
-   g qr0[0], qr2, qr1[0], qr3; // error! qr2 and qr3 differ in size
-
-has a second-to-last line that means
-
-.. code-block:: text
-
-   // FIXME: insert translation of algorithmic block from TeX source.
-
-   for j ← 0, 1 do
-       g qr0[0],qr1[j],qr2[0],qr3[j];
-
-We provide this so that user-defined gates can be applied in parallel
-like the built-in gates.
+and the variable parameters ``params`` can have any type that can promote to ``angle`` type.
 
 Quantum gate modifiers
-----------------------
+~~~~~~~~~~~~~~~~~~~~~~
 
 A gate modifier is a keyword that applies to a gate. A modifier
 :math:`m` transforms a gate :math:`U` to a new gate :math:`m(U)` acting
 on the same or larger Hilbert space. We include modifiers in OpenQASM
 both for programming convenience and compiler analysis.
 
+Control modifiers
++++++++++++++++++
+
 The modifier ``ctrl @`` replaces its gate argument :math:`U` by a
 controlled-:math:`U` gate. If the control bit is 0, nothing happens to the target bit.
 If the control bit is 1, :math:`U` acts on the target bit. Mathematically, the controlled-:math:`U`
 gate is defined as :math:`C_U = I \otimes U^c`, where :math:`c` is the integer value of the control
 bit and :math:`C_U` is the controlled-:math:`U` gate. The new quantum argument is prepended to the
 argument list for the controlled-:math:`U` gate. The quantum argument can be a register, and in this
-case controlled gate broadcast over it (as it shown in examples above for CNOT gate). The modified
+case controlled gate broadcast over it (as for all gates //FIXME). The modified
```

Comment:
What is the plan here?

Reply:
The plan is that all gates broadcast with the same semantics